### ISSUE TITLE
`.plot()` and `.traceplot()` fail with `NameError: name 'subplots' is not defined`

### ISSUE THREAD

**Issue by widdowquinn** [stan-dev/pystan#273]
Plotting `StanFit4Model` parameters with `.plot()` or `.traceplot()` fails with `NameError: name 'subplots' is not defined`

#### Reproducible Steps:
In a Jupyter Notebook:

```
%pylab inline

import numpy as np
import pandas as pd
import pystan
import seaborn as sns

sns.set_context('notebook')

# skipping data import and model setup
[...]

varying_intercept_fit = pystan.stan(model_code=varying_intercept,
                                    data=varying_intercept_data,
                                    iter=1000, chains=2)

# At this point we have a model fit, and type(varying_intercept_fit) returns
# stanfit4anon_model_02868d8446abea90f6f542cf9fa9dd95_6503585146788519231.StanFit4Model

varying_intercept_fit.plot()
```

#### Current Output:
```
# varying_intercept.plot() throws the error below
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-51-ce2dcfc98f24> in <module>()
----> 1 varying_intercept_fit.plot()

stanfit4anon_model_02868d8446abea90f6f542cf9fa9dd95_6503585146788519231.pyx in stanfit4anon_model_02868d8446abea90f6f542cf9fa9dd95_6503585146788519231.StanFit4Model.plot (/var/folders/f9/b9snshhs35sb95fd60kt8x5w0000gn/T/tmpf7qvx3yz/stanfit4anon_model_02868d8446abea90f6f542cf9fa9dd95_6503585146788519231.cpp:10206)()

/usr/local/lib/python3.5/site-packages/pystan/plots.py in traceplot(fit, pars, **kwargs)
     17         logger.critical("matplotlib required for plotting.")
     18         raise
---> 19     return plots.traceplot(fit.extract(), pars, **kwargs)

/usr/local/lib/python3.5/site-packages/pystan/external/pymc/plots.py in traceplot(trace, vars)
     31 
     32     n = len(vars)
---> 33     f, ax = subplots(n, 2, squeeze=False)
     34 
     35     for i, v in enumerate(vars):

NameError: name 'subplots' is not defined
```

#### Expected Output:
I expect a plot of fitted parameters for the model

#### PyStan Version:
2.12.0.0

#### Python Version:
Python 3.5.2 (default, Oct 11 2016, 04:59:56) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)]

Jupyter Notebook server 4.2.0

#### Operating System:
OSX 10.12.1

**Comment by ariddell** [comment]
Thanks for the bug report. I'm not able to reproduce this immediately. The code in plots.py (the file linked to the error) isn't the cleanest. I gather that subplots above is `pylab.subplots()`. I'm not sure why there's a NameError. Perhaps someone else has an insight?

**Comment by ahartikainen** [comment]
Do you have a file named pylab.py in the cwd or somewhere in $PATH?

In jupyter notebook:

    import pylab
    print(pylab)

**Comment by widdowquinn** [comment]
Hi Ari,

No, I do not. Looking at `plots.py`, my suspicion is that the `from pylab import *` isn't playing nicely with other things, and that a reconfiguration of that module file with `import pylab as plt`, and working with the `plt` namespace would likely fix things.

I've not tried/had time to try to fix it myself, though.

L.

**Comment by ahartikainen** [comment]
I can "duplicate" the error if there is a file called pylab.py somewhere where python will first look import.

Pylab should come from "site-packages/pylab.py" and inside that file it does 'from matplotlib.pylab import *'.
matplotlib.pylab does 'from matplotlib.pyplot import *'. So subplots should come from there.

To test where is your pylab you can import pylab module and then print it. And do this in the same folder where the problem occurs.

    import pylab
    print(pylab)

You could check your $PATH with command 'printenv' on terminal.

And finally, what is your matplotlib version and do you use base environment, virtualenv or conda (including conda virtualenv)

**Comment by widdowquinn** [comment]
I can't reproduce the issue locally any more, myself.

After the original failure, I wrote ad hoc functions to visualise the parameter estimates instead, and finished work on the repository a few days later. This is the first time I've returned to it since then. I created a fresh virtualenv from scratch, with new `pip` installations of the relevant packages (`pystan`, `jupyter`, `pandas`, `seaborn`). Rerunning the problem notebook now shows no problem.

I can't be sure that the versions of `matplotlib` etc. I have in this virtualenv are the same as those that were available when I reported the error.

I'd be happy if you closed this with "cannot reproduce".

Many thanks,

L.
